resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/internal/builders/AutoBuildJobTest.java

```diff
@@ -139,11 +155,84 @@ public IProject[] build(int kind, Map<String, String> args, IProgressMonitor mon
 			}
 		});
 
-		// triggers autobuild
-		project.touch(getMonitor());
-		Thread.sleep(Policy.MAX_BUILD_DELAY);
+		triggerAutoBuildAndWait();
 		Job.getJobManager().join(ResourcesPlugin.FAMILY_AUTO_BUILD, null);
 		assertEquals("Should see two scheduled() calls", 2, scheduled.get());
 		assertEquals("Should see two running() calls", 2, running.get());
 	}
-}
+
+	@Test
+	public void testWaitForAutoBuild_JobManagerIsSuspended_ExceptionIsThrown()
+			throws InterruptedException, CoreException, ExecutionException {
+		Job.getJobManager().suspend();
+
+		assertEquals("Scheduled calls", 0, scheduled.get());
+		assertEquals("Running calls", 0, running.get());
+
+		triggerAutoBuildAndWait();
+		Job.getJobManager().join(ResourcesPlugin.FAMILY_AUTO_BUILD, null);
+
+		assertEquals("Scheduled calls", 1, scheduled.get());
+		assertEquals("Running calls", 0, running.get());
+
+		assertThrows(JobManagerSuspendedException.class, () -> waitForAutoBuild(2_000));
+
+		// Job.getJobManager().resume();
```

Comment:
The test currently fails, probably because this line is commented out.

Reply:
Done. I also removed the initial call to `resume` from the other test in 3380fb6a3937f2f406c74ced057a2b653e59a41d since the `JobManager` will always be resumed after each test.